src/exporter/DaemonMetricCollector.cc

```diff
@@ -80,7 +80,22 @@ std::string boost_string_to_std(boost::json::string js) {
 
 std::string quote(std::string value) { return "\"" + value + "\""; }
 
-bool is_hyphen(char ch) { return ch == '-'; }
+void promethize(std::string &name) {
+  auto should_be_underscore = [](char ch) {
+    return ch == '.' || ch == '/' || ch == ' ';
+  };
+  std::replace_if(name.begin(), name.end(), should_be_underscore, '_');
+
+  boost::replace_all(name, "::", "_");
+  boost::replace_all(name, "+", "_plus");
+
+  if (name[name.size() - 1] == '-') {
+    name[name.size() - 1] = '_';
+    name += "minus";
+  } 
+  boost::replace_all(name, "-", "_");
```

Comment:
Can `minus` handling be moved to the top of the method and this (fourth!) iteration be folded into `should_be_underscore` one to reduce the number of iterations and so that `boost::replace_all` is used only for "complex" substitutions?